src/systems/userinput/devices/app-aware-mouse.js

```diff
@@ -1,104 +1,49 @@
-    this.origin = new THREE.Vector3();
-    this.prevOrigin = new THREE.Vector3();
-    this.direction = new THREE.Vector3();
-    this.prevDirection = new THREE.Vector3();
   }
-
   write(frame) {
-    this.prevCursorPose.copy(this.cursorPose);
-    this.prevOrigin.copy(this.origin);
-    this.prevDirection.copy(this.prevDirection);
-
-    if (!this.cursorController) {
-      const rightCursorController = document.getElementById("right-cursor-controller");
-
-      if (rightCursorController && rightCursorController.components) {
-        this.cursorController = rightCursorController.components["cursor-controller"];
-      }
-    }
-
-    if (!this.camera) {
-      this.camera = AFRAME.scenes[0].systems["hubs-systems"].cameraSystem.viewingCamera;
-    }
+    const transformSystem = AFRAME.scenes[0].systems["transform-selected-object"];
 
     const buttonLeft = frame.get(paths.device.mouse.buttonLeft);
     const buttonRight = frame.get(paths.device.mouse.buttonRight);
-    if (buttonLeft && !this.prevButtonLeft && this.cursorController) {
-      const rawIntersections = [];
-      this.cursorController.raycaster.intersectObjects(
-        AFRAME.scenes[0].systems["hubs-systems"].cursorTargettingSystem.targets,
-        true,
-        rawIntersections
-      );
-      const intersection = rawIntersections.find(x => x.object.el);
-      const remoteHoverTarget = intersection && findRemoteHoverTarget(intersection.object);
-      const userinput = AFRAME.scenes[0].systems.userinput;
-      const isInteractable =
-        intersection &&
-        intersection.object.el.matches(
-          ".interactable, .interactable *, .occupiable-waypoint-icon, .teleport-waypoint-icon"
-        );
-      const isPinned =
-        remoteHoverTarget && remoteHoverTarget.components.pinnable && remoteHoverTarget.components.pinnable.data.pinned;
-      const isFrozen = AFRAME.scenes[0].is("frozen");
-      const template =
-        remoteHoverTarget &&
-        remoteHoverTarget.components.networked &&
-        remoteHoverTarget.components.networked.data.template;
-      const isStaticControlledMedia = template && template === "#static-controlled-media";
-      const isStaticMedia = template && template === "#static-media";
-      this.clickedOnAnything =
-        (isInteractable &&
-          (isFrozen || !isPinned) &&
-          (remoteHoverTarget && canMove(remoteHoverTarget)) &&
-          !isStaticControlledMedia &&
-          !isStaticMedia) ||
-        userinput.activeSets.includes(sets.rightCursorHoldingPen) ||
-        userinput.activeSets.includes(sets.rightCursorHoldingInteractable) ||
-        userinput.activeSets.includes(sets.rightCursorHoldingCamera);
-    }
-    this.prevButtonLeft = buttonLeft;
-
-    if (!buttonLeft) {
-      this.clickedOnAnything = false;
-    }
 
-    if ((!this.clickedOnAnything && buttonLeft) || buttonRight) {
+    if ((buttonRight && !transformSystem.transforming) || (buttonLeft && !anyEntityWith(APP.world, HeldRemoteRight))) {
```

Comment:
This got suspiciously simple... I am still not 100% convinced some functionality was not lost here...